extensions/integration-tests/src/notebook.test.ts

```diff
@@ -18,123 +18,160 @@ import * as fs from 'fs';
-				let sparkResult = (<azdata.nb.IStreamResult>cellOutputs[3]).text;
-				assert(sparkResult === '2', `Expected spark result: 2, Actual: ${sparkResult}`);
+				await this.NbTester.PySpark3NbTest(this.test.title);
 			});
 		}
 	});
 }
 
-async function openNotebook(content: azdata.nb.INotebookContents, kernelMetadata: any, testName: string, runAllCells?: boolean): Promise<azdata.nb.NotebookEditor> {
-	let notebookConfig = vscode.workspace.getConfiguration('notebook');
-	notebookConfig.update('pythonPath', getConfigValue(EnvironmentVariable_PYTHON_PATH), 1);
-	let server = await getBdcServer();
-	assert(server && server.serverName, 'No server could be found in openNotebook');
-	await connectToServer(server, 6000);
-	let notebookJson = Object.assign({}, content, { metadata: kernelMetadata });
-	let uri = writeNotebookToFile(notebookJson, testName);
-	console.log(uri);
-	let notebook = await azdata.nb.showNotebookDocument(uri);
-	console.log('Notebook is opened');
-
-	if (!runAllCells) {
-		assert(notebook.document.cells.length === 1, 'Notebook should have 1 cell');
-		console.log('Before run notebook cell');
-		let ran = await notebook.runCell(notebook.document.cells[0]);
-		console.log('After run notebook cell');
-		assert(ran, 'Notebook runCell should succeed');
-	} else {
-		console.log('Before run all notebook cells');
-		let ran = await notebook.runAllCells();
-		assert(ran, 'Notebook runCell should succeed');
-		assert(notebook !== undefined && notebook !== null, 'Expected notebook object is defined');
+class NotebooksTester {
+	setup() {
+		console.log(`environment variable SuiteType set to ${process.env.SuiteType}`);
 	}
 
-	return notebook;
-}
-async function verifyClearAllOutputs(notebook: azdata.nb.NotebookEditor) {
-	let cellWithOutputs = notebook.document.cells.find(cell => cell.contents && cell.contents.outputs && cell.contents.outputs.length > 0);
-	assert(cellWithOutputs !== undefined, 'Could not find notebook cells with outputs');
-	console.log('Before clearing cell outputs');
-	let clearedOutputs = await notebook.clearAllOutputs();
-	let cells = notebook.document.cells;
-	cells.forEach(cell => {
-		assert(cell.contents && cell.contents.outputs && cell.contents.outputs.length === 0, `Expected Output: 0, Actual: '${cell.contents.outputs.length}'`);
-	});
-	assert(clearedOutputs, 'Outputs of all the code cells from Python notebook should be cleared');
-	console.log('After clearing cell outputs');
+	async Cleanup(testName: string) {
+		try {
+			let fileName = getFileName(testName);
+			if (fs.existsSync(fileName)) {
+				fs.unlinkSync(fileName);
+				console.log(`"${fileName}" is deleted.`);
+			}
+			await vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		}
+		catch (err) {
+			console.log(err);
+		}
+		finally {
+			console.log(`"${testName}" is done`);
+		}
+	}
+
+	constructor(testName: string) {
+		console.log(`Start "${testName}"`);
+	}
+
+	async SqlNbTest(testName: string) {
```

Comment:
methods should use `camelCase` method names, so `sqlNbTest`.  Also, probably should add a return type.